planner/core/point_get_plan.go

```diff
@@ -1039,6 +1066,20 @@ func getNameValuePairs(stmtCtx *stmtctx.StatementContext, tbl *model.TableInfo,
 	return nil, false
 }
 
+func checkCanConvertInPointGet(col *model.ColumnInfo, d types.Datum) bool {
+	kind := d.Kind()
+	switch col.FieldType.EvalType() {
+	case ptypes.ETString:
+		switch kind {
+		case types.KindInt64, types.KindUint64,
+			types.KindFloat32, types.KindFloat64, types.KindMysqlDecimal:
```

Comment:
Should `bit` be contained here?

Reply:
I will do bit type test on mysql later.

Reply:
OK, please also test `year` and other int like types.

Reply:
datetime serise is tested, no problem now. do we need test case for this?

Reply:
That's much better. More test cases of different types could help us avoid this problem in the future.

Reply:
@lzmhhh123 `bit` and `Hex binary literal` tested same as MySQL, so just `char` column with `numeric` literal need to be reject from try point get plan.